usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js

```diff
@@ -129,38 +358,59 @@
 
     this.leftMeetingZone = function(id, params) {
         if (that.entityID === id) {    
-            var displayName = params[0];
 
-            var index = that.room.occupants.indexOf(displayName);
-            if (index > -1) {
-                that.room.occupants.splice(index, 1);
-            }
+            var uuid = params[0];
 
+            if (uuid in that.room.occupants) {
+                delete that.room.occupants[uuid];
+            }
+            var text = Object.keys(that.room.occupants).map(function(key) {
+                return that.room.occupants[key];
+            });
             Entities.editEntity(id, {
-                "text": (that.room.occupants).join("\n"), 
+                "text": text.join("\n"), 
                 "textColor": [255, 255, 255]
             });
         }
     };  
     
     
     this.updateSignColor = function(id, params) {
-        if (that.entityID === id) {    
-            if (params[1] === "GREEN") {
+        if (id === that.roomEntityIDs[1]) {    
+            if (params[0] === "GREEN") {
```

Comment:
please don't parse your colors by matching strings contained within a single-value array argument to this function.

the best way i can see to do this would be to have `updateSignColor` take a bool like `isOccupied` as its second parameter. then your conditionals could operate on the flag status instead of string matching